react/features/calendar-sync/components/MeetingList.native.js

```diff
@@ -0,0 +1,237 @@
+// @flow
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+
+import { refreshCalendarEntryList } from '../actions';
+
+import { appNavigate } from '../../app';
+import { getLocalizedDateFormatter, translate } from '../../base/i18n';
+import { NavigateSectionList } from '../../base/react';
+
+type Props = {
+
+    /**
+     * Indicates if the list is disabled or not.
+     */
+    disabled: boolean,
+
+    /**
+     * The Redux dispatch function.
+     */
+    dispatch: Function,
+
+    /**
+     * Tells the component if it's being displayed at the moment, or not.
+     * Note: as an example, on Android it can happen that the component
+     * is rendered but not displayed, because components like ViewPagerAndroid
+     * render their children even if they are not visible at the moment.
+     */
+    displayed: boolean,
+
+    /**
+     * The calendar event list.
+     */
+    _eventList: Array<Object>,
+
+    /**
+     * The translate function.
+     */
+    t: Function
+};
+
+/**
+ * Component to display a list of events from the (mobile) user's calendar.
+ */
+class MeetingList extends Component<Props> {
+    _initialLoaded: boolean
+
+    /**
+     * Constructor of the MeetingList component.
+     *
+     * @inheritdoc
+     */
+    constructor(props) {
+        super(props);
+
+        this._onPress = this._onPress.bind(this);
+        this._onRefresh = this._onRefresh.bind(this);
+        this._toDisplayableItem = this._toDisplayableItem.bind(this);
+        this._toDisplayableList = this._toDisplayableList.bind(this);
+        this._toDateString = this._toDateString.bind(this);
+    }
+
+    /**
+     * Implements React Component's componentWillReceiveProps function.
+     *
+     * @inheritdoc
+     */
+    componentWillReceiveProps(newProps) {
+        // This is a conditional logic to refresh the calendar entries (thus
+        // to request access to calendar) on component first receives a
+        // displayed=true prop - to avoid requesting calendar access on
+        // app start.
+        if (!this._initialLoaded
+                && newProps.displayed
+                && !this.props.displayed) {
+            const { dispatch } = this.props;
+
+            this._initialLoaded = true;
+            dispatch(refreshCalendarEntryList());
```

Comment:
What do you think of having the parent of MeetingList, the one that passes displayed, refresh the calendar list instead? That way this component is focused on display. As always, feel free to disagree; I'm offering my musings.

Reply:
I can move this part of the logic out of the component (e.g. into the PagedList component) but then a similar refresh logic would still stay in the component, that runs on pull-to-refresh. What do you think?

Reply:
Eh. If I felt strongly I'd mark the pr as changes requested. Whatever you think is best, then, works for me, as you seem to have a good reason for having the current logic as is, and I'm okay with the current logic.